### PR TITLE
Fixed issue where clusterAddressMap did not remove this instance after the instance was offline

### PR DESCRIPTION
**Pull request by wanghuizuo**
### Ⅰ. Describe what this PR did
1. Fixed issue where clusterAddressMap did not remove this instance after the instance was offline.
2. remove EurekaRegistryException, because EurekaRegistryException is RuntimeException.
3. add log.

### CODE REVIEW

File: discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java [hunk 1]
```diff
@@ -15,36 +15,27 @@
...
-import java.util.Properties;
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-
+import com.netflix.appinfo.ApplicationInfoManager;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.providers.EurekaConfigBasedInstanceInfoProvider;
+import com.netflix.config.ConfigurationManager;
+import com.netflix.discovery.*;
```
Comment (xingfudeshi) [inline discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: please do not import *

File: discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java [hunk 2]
```diff
@@ -15,36 +15,27 @@
...
-import com.netflix.discovery.EurekaClient;
-import com.netflix.discovery.EurekaEvent;
-import com.netflix.discovery.EurekaEventListener;
-import com.netflix.discovery.shared.Application;
-import com.netflix.discovery.shared.Applications;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.net.InetSocketAddress;
+import java.util.*;
```
Comment (xingfudeshi) [inline discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: Please do no import *.

File: discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java [hunk 3]
```diff
@@ -140,7 +131,7 @@ public void onEvent(EurekaEvent event) {
                     try {
                         refreshCluster();
                     } catch (Exception e) {
-                        LOGGER.error("Eureka event listener refreshCluster error!");
+                        LOGGER.error("Eureka event listener refreshCluster error!", e);
```
Comment (xingfudeshi) [inline discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: LOGGER.error("Eureka event listener refreshCluster error:{}", e.getMessage(),e);

File: discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java [hunk 4]
```diff
@@ -157,25 +148,37 @@ public void close() throws Exception {
...
-        for (Application app : list) {
-            Set<InetSocketAddress> addressSet = new HashSet<>();
-            List<InstanceInfo> instances = app.getInstances();
-            if (CollectionUtils.isNotEmpty(instances)) {
-                for (InstanceInfo instance : instances) {
-                    addressSet.add(new InetSocketAddress(instance.getIPAddr(), instance.getPort()));
-                }
+
+        int expectedSize = applications.size();
+        int initialCapacity = expectedSize < 3? expectedSize + 1: (int) ((float) expectedSize / 0.75F + 1.0F);
```
Comment (xingfudeshi) [inline discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: You can directly use 8 here.And the calculation for capacity may not required.

Comment (xingfudeshi) [reply discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: Please define a constant,like 
```
private static final int INITIAL_CAPACITY = 8;
```

File: discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java [hunk 5]
```diff
@@ -157,25 +148,37 @@ public void close() throws Exception {
...
+        ConcurrentMap<String, Set<InetSocketAddress>> collect = new ConcurrentHashMap<>(initialCapacity);
+
+        for (Application application : applications) {
+            List<InstanceInfo> instances = application.getInstances();
+            Set<InetSocketAddress> addressSet = new HashSet<>(instances.size());
```
Comment (xingfudeshi) [inline discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: Use INITIAL_CAPACITY,the value of instance.size() may not be the power of 2.

File: discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java [hunk 6]
```diff
@@ -157,25 +148,37 @@ public void close() throws Exception {
...
+            for (InstanceInfo instance : instances) {
+                addressSet.add(new InetSocketAddress(instance.getIPAddr(), instance.getPort()));
             }
-            clusterAddressMap.put(app.getName(), addressSet);
+            collect.put(application.getName(), addressSet);
```
Comment (slievrly) [inline discovery/seata-discovery-eureka/src/main/java/io/seata/discovery/registry/eureka/EurekaRegistryServiceImpl.java]: if instances is empty why put?

**Pull request merged**